Redo vPath()

bradyakent opened the pull request:
`tex.vPath()`'s  return values' meanings have changed:

- `"right"` now means "turn right, move forward" rather than "turn right, move forward, turn right" 
- `"left"` now means "turn left, move forward" rather than "turn left, move forward, turn left"

This allows programs and APIs that use `tex.vPath()`'s return format to be interfaced with easier, such as the tprint API. The goal of this is to allow me to make programs that write more optimized tcode, because as of right now, `tprint.scan()`'s output requires turtles to move through every block in a volume to print the tcode's structure.

bradyakent committed:
vPath() "right" and "left" meaning changed

"right" and "left" now means "turn once, then move forward", not "turn, move forward, turn again"; allows me to write more optimized tcode and other iterators that return the same values.

bradyakent committed:
update code that references vPath()

Since vPath()'s usage has changed, code that references it has been updated to reflect this.

BradyFromDiscord commented:
vPath() may have some issues, I'm going to go test it real quick, but I suspect the following changes present a problem:
- some of the state variables aren't being updated when they were before.
- when the turtle turns right, then turns right a second time, dt isn't being incremented even though the turtle is travelling along the depth axis.

bradyakent committed:
fix dt and wt resets

one of the concerns raised in #12

bradyakent committed:
increment dt on shift

dt must be incremented any time the turtle travels along the depth axis. Fixes concern from #12

bradyakent committed:
Write comments for vPath()

Help me understand what's going on if I need to revisit this.
